lms/djangoapps/instructor_task/tasks_helper/misc.py

```diff
@@ -301,3 +303,126 @@ def upload_ora2_data(
     TASK_LOG.info(u'%s, Task type: %s, Upload complete.', task_info_string, action_name)
 
     return UPDATE_STATUS_SUCCEEDED
+
+
+def upload_students_grades_data(
+        _xmodule_instance_args, _entry_id, course_id, _task_input, action_name
+):
+    """
+    Collect student grades and upload them to S3 as a CSV.
+    """
+
+    headers = ['email', 'cohort']
+
+    students = User.objects.filter(
+        courseenrollment__course_id=course_id,
+        courseenrollment__is_active=1,
+    ).order_by('id').prefetch_related('course_groups')
+
+    output = list()
+
+    # Log information
+    start_date = datetime.now(UTC)
+    start_time = time()
+
+    num_attempted = 1
+    num_total = 1
+
+    fmt = u'Task: {task_id}, InstructorTask ID: {entry_id}, Course: {course_id}, Input: {task_input}'
+    task_info_string = fmt.format(
+        task_id=_xmodule_instance_args.get('task_id') if _xmodule_instance_args is not None else None,
+        entry_id=_entry_id,
+        course_id=course_id,
+        task_input=_task_input
+    )
+    TASK_LOG.info(u'%s, Task type: %s, Starting task execution', task_info_string, action_name)
+
+    task_progress = TaskProgress(action_name, num_total, start_time)
+    task_progress.attempted = num_attempted
+
+    curr_step = {'step': "Collecting responses"}
+    TASK_LOG.info(
+        u'%s, Task type: %s, Current step: %s for all submissions',
+        task_info_string,
+        action_name,
+        curr_step,
+    )
+
+    task_progress.update_task_state(extra_meta=curr_step)
+
+    try:
+        for student in students:
+            # Check student access
+            course = get_course_with_access(student, 'load', course_id, check_if_enrolled=True)
+
+            # Get students grade
+            course_grade = CourseGradeFactory().read(student, course)
```

Comment:
Could you move the end of the try block here and set less generic exception, if it possible(i.e. BigGradeException instead Exception)